web/src/features/Dashboard/components/DashboardSnapshotsCard.tsx

```diff
@@ -279,131 +148,175 @@ class DashboardSnapshotsCard extends React.Component<Props, State> {
     }
 
     // if nothing exists yet, we've determined default state is good
-    this.setState({
+    setState({
       determiningDestination: false,
       selectedDestination: find(DESTINATIONS, ["value", "aws"]),
     });
   };
 
-  componentDidMount() {
-    this.fetchSnapshotSettings();
-    if (this.state.snapshotSettings) {
-      this.setCurrentProvider();
-    }
-  }
+  const onSnapshotSettingsSuccess = (result: SnapshotSettings) => {
+    setState({
+      kotsadmRequiresVeleroAccess: false,
+      isLoadingSnapshotSettings: false,
+      snapshotSettingsErr: false,
+      snapshotSettingsErrMsg: "",
+    });
 
-  componentDidUpdate(
-    _lastProps: Props,
-    lastState: { snapshotSettings: SnapshotSettings | null }
-  ) {
-    if (
-      this.state.snapshotSettings !== lastState.snapshotSettings &&
-      this.state.snapshotSettings
-    ) {
-      this.setCurrentProvider();
+    setCurrentProvider(result);
+  };
+  const onSnapshotSettingsError = (err: Error) => {
+    setState({
+      isLoadingSnapshotSettings: false,
+      snapshotSettingsErr: true,
+      snapshotSettingsErrMsg: err.message,
+    });
+  };
+
+  const { data: snapshotSettings } = useSnapshotSettings(
+    onSnapshotSettingsSuccess,
+    onSnapshotSettingsError
+  );
+  const previousSnapshotSettings = usePrevious(snapshotSettings);
+
+  const onCreateSnapshotSuccess = (data: {
+    success?: boolean;
+    option?: "full" | "partial" | undefined;
+  }) => {
+    setState({
+      startingSnapshot: false,
+    });
+    ping();
+    if (data.option === "full") {
+      history.push("/snapshots");
+    } else {
+      history.push(`/snapshots/partial/${app.slug}`);
     }
-  }
+  };
+
+  const onCreateSnapshotError = (err: Error) => {
+    setState({
+      startingSnapshot: false,
+      startSnapshotErr: true,
+      startSnapshotErrorMsg: err.message,
+    });
+  };
+
+  const { mutate: createSnapshot } = useCreateSnapshot(
+    onCreateSnapshotSuccess,
+    onCreateSnapshotError
+  );
+
+  const toggleSnaphotDifferencesModal = () => {
+    setState({
+      snapshotDifferencesModal: !state.snapshotDifferencesModal,
+    });
+  };
 
-  render() {
-    const { isSnapshotAllowed } = this.props;
-    const { selectedDestination } = this.state;
+  useEffect(() => {
+    if (snapshotSettings !== previousSnapshotSettings && snapshotSettings) {
+      setCurrentProvider(snapshotSettings);
+    }
+  }, []);
+  /// useEffects /////
 
-    return (
-      <div className="flex-column flex1 dashboard-card">
```

Comment:
everything below here has not been changed. Reformatting was done because this was converted into a functional component.